rest/main_legacy.go

```diff
@@ -21,12 +22,19 @@ func legacyServerMain(osArgs []string) error {
 		LegacyServerConfig:      *config,
 	}
 
-	sc, databases, err := lc.ToStartupConfig()
+	sc := DefaultStartupConfig(defaultLogFilePath)
+
+	migratedStartupConfig, databases, err := lc.ToStartupConfig()
+	if err != nil {
+		return err
+	}
+
+	err = mergo.Merge(&sc, migratedStartupConfig, mergo.WithOverride)
```

Comment:
Are there any scenarios where DefaultStartupConfig sets a non-zero/non-empty value, and setting that to an empty value in migratedStartupConfig is valid?  Are we guarding against that by defining all such values as pointers?  (my other comment may be related to this)